src/sentry/db/postgres/roles.py

```diff
@@ -6,6 +6,8 @@
 
 from django.db.transaction import get_connection
 
+from sentry.silo.patches.silo_aware_transaction_patch import determine_using_by_silo_mode
+
 
 @contextlib.contextmanager
 def in_test_psql_role_override(role_name: str, using: str | None = None):
```

Comment:
You'll notice in this PR that I initially hunted down the `in_test_psql_role_override` callsites and forced them to provide using explicitly.  It may be noisy -- if we want to lean into just using the default `using=` logic below and not require specification, we can remove those from this PR.

Reply:
I think being explicit about `using` is a good precedent to set. Being explicit about what connection queries need to be run on helps both hybrid cloud and expanding postgres primaries in the future.